getting_started_modules.md

```diff
@@ -0,0 +1,171 @@
+
+#### 4. Using a Terraform Module
+At this point we have created three files (or one): 
+
+- `main.tf` 
+- `variables.tf` 
+- `outputs.tf` 
+
+Together they do one thing: Start a nomad job. Aren't we done with our goal then? Almost. We still need to verify that it works the way we expect it to. Next step is to write some terraform code that runs the module we just made. Then we'll import that code to our virtual machine and run it there. There are two steps to this
+1. Write the code that uses the module, and run it manually inside the virtual machine 
+2. Write ansible code that'll run it automatically when you start the box.
+
+Below is an example on how to use a module:
+
+ ```hcl-terraform
+module "whatever" {
+  source      = "github.com/fredrikhgrelland/terraform-nomad-minio.git?ref=0.0.3"
+}
+```
+
+This will fetch the module at the given source. In the case above it is a MinIO module, version 0.0.3. 
+
+
+##### a. How To Use The Module and Run It Manually
+Create a file called `main.tf` under the `example/` directory, and add the code above, change the module's name, and source. The source is `../` in this case because that is where our module files are, relative to the `main.tf` you are writing this in. 
+
+Let's log onto our virtual machine and try and run it! Run `vagrant ssh` if needed, and navigate to your `/vagrant/example/` folder. Next run `terraform init` to initialize a terraform-workspace. When that is done you can try `terraform plan`, which will read your terraform code and attempt to make a runtime plan. When doing so you will get the error "Error: Missing required argument The argument "address" is required, but was not set.". This is because we are using a resource that deploys a nomad job, but nowhere in our terraform files have we defined _ what_ Nomad to use. At the moment no Nomad is known. This is where [providers]() come into the picture. They are providers for the resources we are using, and in our case we need to define a [Nomad provider](). We could do this in either of our `main.tf` files, but if we do it in our module's `main.tf` it will be very difficult for anyone to use our module, because the Nomad's address is predefined. Instead we should include what nomad to use in our example´s `main.tf`, which is simply an example of how to use the module, meaning anyone else wanting to use the module could supply their own nomad when using the module. To supply a provider add the lines below to your `example/main.tf` file:
```

Comment:
That's tough to review one-line text. Could you split line by line sentences?
```suggestion
Let's log onto our virtual machine and try and run it! Run `vagrant ssh` if needed, and navigate to your `/vagrant/example/` folder. Next run `terraform init` to initialize a terraform-workspace. When that is done you can try `terraform plan`, which will read your terraform code and attempt to make a runtime plan. When doing so you will get the error "Error: Missing required argument The argument "address" is required, but was not set.". This is because we are using a resource that deploys a nomad job, but nowhere in our terraform files have we defined _what_ Nomad to use. At the moment no Nomad is known. This is where [providers](https://www.terraform.io/docs/providers/index.html) come into the picture. They are providers for the resources we are using, and in our case we need to define a [Nomad provider](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs). We could do this in either of our `main.tf` files, but if we do it in our module's `main.tf` it will be very difficult for anyone to use our module, because the Nomad's address is predefined. Instead we should include what nomad to use in our example´s `main.tf`, which is simply an example of how to use the module, meaning anyone else wanting to use the module could supply their own nomad when using the module. To supply a provider add the lines below to your `example/main.tf` file:
```

Reply:
I've split all lines in modules getting started